src/udp_wrap.cc

```diff
@@ -284,12 +296,16 @@ void UDPWrap::DoSend(const FunctionCallbackInfo<Value>& args, int family) {
   if (err == 0) {
     err = uv_udp_send(&req_wrap->req_,
```

Comment:
It's not switching, the idea is to use it to try to send the datagram on the spot, because the socket *might* be writable. Have a look at how `uv_try_write` is used in stream_wrap.cc.